### PR TITLE
feat: Option to remove WorkspaceExec from `owner` role

### PR DESCRIPTION
**Pull request by Emyrk**
# What this does

This provides an option to remove `WorkspaceExec` from the owner role, preventing access to all workspaces.

# How it does this.

I removed the `Wildcard` option from our RBAC. So we have to explicitly list all permissions. I made helpers to make this easy. The wildcard was out of laziness, this allows more granularity.

# Slight Annoyance

Our builtin static roles are global. So to update the roles to remove a permission, I had to make a function `ReloadBuiltinRoles` that reloads the static roles. This has to be set on each coderd, as authz is not synced across the database.

### CODE REVIEW

File: scripts/rbacgen/main.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package main
```
Comment (Emyrk) [inline scripts/rbacgen/main.go:1]: I really want to make sure we keep this enum synced. We should probably make a more generic way to maintain enum lists.

File: cli/testdata/coder_server_--help.golden [hunk 2]
```diff
@@ -16,6 +16,12 @@ Start a Coder server
           $CACHE_DIRECTORY is set, it will be used for compatibility with
           systemd.
 
+      --disable-owner-workspace-exec bool, $CODER_DISABLE_OWNER_WORKSPACE_EXEC
```
Comment (coadler) [inline cli/testdata/coder_server_--help.golden:19]: Maybe `--disable-owner-workspace-access` would be a bit clearer? I'm thinking maybe bc it's not just disabling shell access but apps as well.

**Review by coadler — APPROVED** [review]

**Review by johnstcn — APPROVED** [review]
100% in favour of the autogen stuff. :+1:

I'm slightly more wary of the path we're starting to go down where we have a fairly well-defined set of roles / permissions etc. that we are now having to provide options to customise in various ways. Allowing complete configurability of this would lead to a combinatorial explosion of options. For this small tightly-scoped change it's fine, but we may have to think about how we expose custom RBAC stuff down the line.

**Comment by Emyrk** [comment]
> 100% in favour of the autogen stuff. +1
> 
> I'm slightly more wary of the path we're starting to go down where we have a fairly well-defined set of roles / permissions etc. that we are now having to provide options to customise in various ways. Allowing complete configurability of this would lead to a combinatorial explosion of options. For this small tightly-scoped change it's fine, but we may have to think about how we expose custom RBAC stuff down the line.

Yea, I think if customization become more common we would have to persist roles differently.

**Pull request merged**